### PR TITLE
Upgrade to TYPO3 version 9

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: Classes/SecurityService.php [hunk 1]
```diff
@@ -37,7 +49,9 @@ class SecurityService
      */
     public function __construct()
     {
-        $this->database = $GLOBALS['TYPO3_DB'];
+        $this->connection = GeneralUtility::makeInstance(ConnectionPool::class);
+        $this->keysQueryBuilder = $this->connection->getQueryBuilderForTable('tx_t3amserver_keys');
```
Comment (vertexvaar) [inline Classes/SecurityService.php]: `QueryBuilders` are **never** reused, so they are instantiated for each query

File: Classes/SecurityService.php [hunk 2]
```diff
@@ -37,7 +49,9 @@ class SecurityService
      */
     public function __construct()
     {
-        $this->database = $GLOBALS['TYPO3_DB'];
+        $this->connection = GeneralUtility::makeInstance(ConnectionPool::class);
```
Comment (vertexvaar) [inline Classes/SecurityService.php]: You are saving an instance of `ConnectionPool` into a property named `connection`. You should rename the property.

File: Classes/UserRepository.php [hunk 3]
```diff
@@ -59,7 +67,8 @@ class UserRepository
      */
     public function __construct()
     {
-        $this->connection = $GLOBALS['TYPO3_DB'];
+        $this->connection = GeneralUtility::makeInstance(ConnectionPool::class);
```
Comment (vertexvaar) [inline Classes/UserRepository.php]: Same here: Rename property to connection pool

File: Classes/UserRepository.php [hunk 4]
```diff
@@ -59,7 +67,8 @@ class UserRepository
...
+        $this->beUserQueryBuilder = $this->connection->getQueryBuilderForTable('be_users');
```
Comment (vertexvaar) [inline Classes/UserRepository.php]: Same here: Do not reuse query builders

File: Classes/SecurityService.php [hunk 5]
```diff
@@ -49,7 +63,7 @@ public function processDatamap_beforeStart(DataHandler $dataHandler)
             foreach (array_keys($dataHandler->datamap['tx_t3amserver_client']) as $uid) {
                 if (is_string($uid) && 0 === strpos($uid, 'NEW')) {
                     $dataHandler->datamap['tx_t3amserver_client'][$uid]['token'] = GeneralUtility::hmac(
-                        GeneralUtility::generateRandomBytes(256),
+                        (new Random)->generateRandomBytes(256),
```
Comment (vertexvaar) [inline Classes/SecurityService.php]: Please use `GeneralUtility::makeInstance(Random::class)` instead

File: Classes/UserRepository.php [hunk 6]
```diff
@@ -1,4 +1,5 @@
 <?php
+
```
Comment (vertexvaar) [inline Classes/UserRepository.php]: do not add empty lines before namespaces

File: Classes/UserRepository.php [hunk 7]
```diff
@@ -89,8 +122,33 @@ public function getUserState($user)
      */
     public function getUser($user)
     {
-        $where = 'username = ' . $this->connection->fullQuoteStr($user, 'be_users');
-        return $this->connection->exec_SELECTgetSingleRow(implode(',', $this->fields), 'be_users', $where);
+        foreach ($this->fields as $field) {
+            $this->beUserQueryBuilder
+                ->addSelect($field);
```
Comment (vertexvaar) [inline Classes/UserRepository.php]: You can use the splat operator instead:
```
$queryBuilder->select(...$this->fields)
             ->fr...
```

File: composer.json [hunk 8]
```diff
@@ -13,7 +13,7 @@
     "require": {
         "php": "^5.5 || ^7.0",
         "ext-openssl": "*",
-        "typo3/cms-core": "^7.6 || ^8.7"
+        "typo3/cms-core": "^9.5"
```
Comment (vertexvaar) [inline composer.json]: we have the habit of supporting the current two LTS versions of TYPO3 and i think most, if not all, changes are compatible to TYPO3 v8. That version should be included in this list.

**Review by vertexvaar — CHANGES_REQUESTED** [review]
Hi @dejorid 

thank you very much for your pull request :+1: . I've seen you created a new github account (probably especially for this?) so welcome to github, too :tada: .
We really appreciate your changes but require you to fix the one or other thing, before i can merge your changes.
I've added comments to the lines that need a change (exemplary, not exhaustive).

One thing that affects all files: We follow the PSR-2 codestyle. I am pretty sure your IDE does support code reformating and the PSR-2 preset. Please reformat your code accordingly. This is not required for you to get this merged. If you don't do it, i will :wink: 

With Regards

**Comment by ghost** [comment]
Hello @vertexvaar. Thank you for reviewing my pull request that quickly! :grinning: I will apply the requested changes as soon as possible.

**Comment by ghost** [comment]
Now all the issues should be resolved! Thanks again for your review. Let me know if there is anything left to be changed.

**Comment by ghost** [comment]
I fixed some additional issues I found in the code. Everything should be working fine now.

**Comment by vertexvaar** [comment]
Hi @dejorid (@dejdav ?)

i am now going to review your changes again.

**Comment by ghost** [comment]
Thanks a lot!

**Comment by vertexvaar** [comment]
@dejorid Sorry this took longer, some high prio stuff landed on my desk. Nevertheless you did a great job. Thank you very much for your PR! I will create a release after some cleanups :beers:

**Comment by ghost** [comment]
Thanks for approving the pull request. 👍

**Pull request merged**